prow/cmd/checkconfig/main.go

```diff
@@ -793,24 +848,121 @@ func validateNeedsOkToTestLabel(cfg *config.Config) error {
 	return errorutil.NewAggregate(queryErrors...)
 }
 
-func verifyOwnersPlugin(cfg *plugins.Configuration) error {
+type filePresenceChecker interface {
+	remoteCheck(path string) (bool, error)
+	localCheck(path string, client *git.Client) bool
+}
+
+type ghRepo struct {
+	full, org, repo string
+}
+
+func (f ghRepo) remoteCheck(path string) (bool, error) {
+	blobUrl := github.BlobUrl(f.org, f.repo, "master", path)
+	resp, err := http.Head(blobUrl)
+	if err != nil {
+		return false, err
+	}
+	return resp.StatusCode == 200, nil
+}
+
+func (f ghRepo) localCheck(path string, gc *git.Client) bool {
+	r, err := gc.Clone(f.full)
+	if err != nil {
+		return false
+	}
+
+	// checkconfig is not a long-running process and can be often run locally,
+	// so we would not gain much by keeping the cache
+	defer r.Clean()
+	_, err = os.Stat(filepath.Join(r.Dir, path))
+
+	return err == nil
```

Comment:
Isn't there a specific error for not existing? Other errors mean we failed to touch the filesystem and should quit the process, maybe?